crates/ruff_formatter/src/arguments.rs

```diff
@@ -33,23 +33,26 @@ impl<'fmt, Context> Argument<'fmt, Context> {
     #[doc(hidden)]
     #[inline]
     pub fn new<F: Format<Context>>(value: &'fmt F) -> Self {
+        #[allow(clippy::inline_always)]
         #[inline(always)]
         fn formatter<F: Format<Context>, Context>(
             ptr: *const c_void,
             fmt: &mut Formatter<Context>,
         ) -> FormatResult<()> {
             // SAFETY: Safe because the 'fmt lifetime is captured by the 'lifetime' field.
-            F::fmt(unsafe { &*(ptr as *const F) }, fmt)
+            #[allow(unsafe_code)]
+            F::fmt(unsafe { &*ptr.cast::<F>() }, fmt)
         }
 
         Self {
-            value: value as *const F as *const c_void,
+            value: (value as *const F).cast::<std::ffi::c_void>(),
             lifetime: PhantomData,
             formatter: formatter::<F, Context>,
         }
     }
 
     /// Formats the value stored by this argument using the given formatter.
+    #[allow(clippy::inline_always)]
```

Comment:
The docs says (https://rust-lang.github.io/rust-clippy/v0.0.212/index.html#inline_always):

> While there are valid uses of this annotation (and once you know when to use it, by all means allow this lint), it’s a common newbie-mistake to pepper one’s code with it. As a rule of thumb, before slapping #[inline(always)] on a function, measure if that additional function call really affects your runtime profile sufficiently to make up for the increase in compile time.

Do we have measurements that `#[inline(always)]` is better than `#[inline]`?